test/sys/test_socket.rs

```diff
@@ -1462,3 +1462,38 @@ pub fn test_vsock() {
     close(s1).unwrap();
     thr.join().unwrap();
 }
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[test]
+pub fn test_sockaddr_storage_to_datalink() {
+    use nix::sys::socket::sockaddr_storage;
+    // in C, sockaddr_* are as if sockaddr_storage was a union
+    let mut storage : sockaddr_storage = unsafe { std::mem::zeroed::<sockaddr_storage>() };
+    let mut ll = unsafe { &mut *(&mut storage as *mut _ as *mut libc::sockaddr_ll) };
+    // NOTE: libc has libc::AF_PACKET as cint, instead of u16
+    ll.sll_family = libc::AF_PACKET as u16;
+    // NOTE: libc has libc::ETH_P_IP as cint, instead of u16
+    ll.sll_protocol = (libc::ETH_P_IP as u16).to_be();
+    ll.sll_ifindex = 1i32;
+    ll.sll_hatype = libc::ARPHRD_ETHER.to_be();
+    ll.sll_pkttype = 0u8; // PACKET_HOST (constants not in libc crate)
+    ll.sll_halen = 6;
+    // 00:00:5E:53:00 is reserved for documentation.
+    ll.sll_addr = [0x02, 0x00, 0x5E, 0x00, 0x53, 0x00, 0x00, 0x00];
```

Comment:
Are you talking about this:
>      00-00-5E-00-53-00 through 00-00-5E-00-53-FF: assigned for use in
>         documentation.
https://tools.ietf.org/html/rfc7042#section-2.1.1

Reply:
That leading `0x02` should be a `0x00`, right?

Reply:
Ah, I was very careful about transcribing the array, and screwed up the comment!  Thanks for noticing that. Fixed.

MAC addresses are sent on the wire Least Significant Bit first.  So the first few bits (lowest few bits) of the mac have special meanings.  0x01 means it's a "group" address (used by multicast or broadcast packets), and 0x02 means "Not globally unique."

Flipping 0x00 to 0x02 is probably not necessary here (people know that the documentation prefixes are not globally unique), but it's habit I have when I'm manually assigning addresses.

The 0x00 padding at the end is annoying, but necessary based on the types.